erpnext-openmrs/src/main/java/com/ozonehis/eip/erpnext/openmrs/routes/PatientRouting.java

```diff
@@ -31,7 +31,7 @@ public class PatientRouting extends RouteBuilder {
 
     Predicate isPatientSyncEnabled() {
         return exchange -> isPatientSyncEnabled
-                || Boolean.TRUE.equals(exchange.getIn().getHeader(HEADER_ENABLE_PATIENT_SYNC, Boolean.class));
+                || exchange.getMessage().getHeader(HEADER_ENABLE_PATIENT_SYNC, false, Boolean.class);
     }
```

Comment:
How about adding the logic here, so that `isPatientSyncEnabled` method is the single source of truth for checking if patient sync is enabled or not. WDYT @corneliouzbett ?

```
  Predicate isPatientSyncEnabled() {
        return exchange -> isPatientSyncEnabled
                || exchange.getMessage().getHeader(HEADER_ENABLE_PATIENT_SYNC, false, Boolean.class)
                || "u".equals(exchange.getMessage().getHeader(HEADER_FHIR_EVENT_TYPE, "c", String.class));
    }
```

Reply:
That make more sense. Thanks @VaishSiddharth!